pkg/peer/subscription/rtp.go

```diff
@@ -0,0 +1,135 @@
+package subscription
+
+import (
+	"fmt"
+
+	"github.com/pion/rtp"
+	"golang.org/x/exp/constraints"
+)
+
+type RewrittenRTPPacket *rtp.Packet
+
+// A structure that is used to rewrite the RTP packets that are being forwarded.
+type PacketRewriter struct {
+	// The SSRC of the previously forwarded packet.
+	previouslyForwardedSSRC uint32
+	// SSRC that we're using for all packets that we're forwarding.
+	// This is the SSRC that we're sending to the remote peer. Typically,
+	// this is the SSRC of the lowest layer for the simulcast track.
+	outgoingSSRC uint32
+	// The highest identifiers of the forwarded packet,i.e. the IDs of
+	// the **latest** (in terms of timestamp and number) packet. This is not
+	// necessarily the IDs of the **last forwarded packet**, due to packets
+	// that may arrive out-of-order.
+	maxOutgoingIDs PacketIdentifiers
```

Comment:
While it might not be exactly the last forwarded packet, I still think `lastOutgoingIDs` would be more descriptive and give you a better picture when reading the code for the first time

Reply:
Personally, I prefer `last` even if slightly wrong... Does anyone else have any opinions on this?

Reply:
The thing is... I might need to use `last` as the last packet that we received when properly handling wrapping, so this name would be occupied 🙂 What about `latest`? Maybe sounds more reasonable than `last` and `max`? (I think I might actually indeed prefer `latest`)

Reply:
`latest` sounds great!